.circleci/scripts/trigger_azure_pipeline.py

```diff
@@ -0,0 +1,99 @@
+import os
+import re
+import sys
+import requests
+import time
+
+AZURE_PIPELINE_BASE_URL = "https://aiinfra.visualstudio.com/PyTorch/"
+AZURE_DEVOPS_PAT_BASE64 = "LW46YnZoNWhwM2xnZ3MzdWVqM2VteDR0ZnV4cHJ6djVwc2xuNWg0emZqaDdsa2Y0ZjdzN2lzcQ=="
+PIPELINE_ID = "911"
+PROJECT_ID = "0628bce4-2d33-499e-bac5-530e12db160f"
+
+s = requests.Session()
+s.headers.update({"Authorization": "Basic " + AZURE_DEVOPS_PAT_BASE64})
+
+build_base_url = AZURE_PIPELINE_BASE_URL + "_apis/build/builds?api-version=6.0"
+
+TARGET_BRANCH = os.environ.get("CIRCLE_BRANCH", "master")
+
+# Convert the branch name for Azure DevOps
+if TARGET_BRANCH.startswith('pull/'):
+    match = re.search(r'pull/(\d+)', TARGET_BRANCH)
```

Comment:
Let's make code a bit more error resilient
```suggestion
match = re.search(r'pull/(\d+)', TARGET_BRANCH)
if match is not None:
```